pkg/client/daemon/outbound.go

```diff
@@ -135,18 +137,40 @@ func (o *outbound) routerServerWorker(c context.Context) (err error) {
 // in the search path declared in the Docker config. The "tel2-search" domain fills this purpose and a request for
 // "<single label name>.tel2-search." will be resolved as "<single label name>." using the search path of this resolver.
 const tel2SubDomain = "tel2-search"
-const dotTel2SubDomain = "." + tel2SubDomain
-const dotKubernetesZone = "." + kubernetesZone
+const dotTel2SubDomain = "." + tel2SubDomain + "."
+const dotKubernetesZone = "." + kubernetesZone + "."
 
 var localhostIPv6 = []net.IP{{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1}}
 var localhostIPv4 = []net.IP{{127, 0, 0, 1}}
 
+func (o *outbound) doClusterLookup(query string) bool {
+	if strings.HasSuffix(query, dotKubernetesZone) && strings.Count(query, ".") < 4 {
```

Comment:
Why `strings.Count(query, ".") < 4`?

Reply:
I guess that's sorta answered by the old code
```patch
-	// We don't care about queries to the kubernetes zone unless they have at least two additional elements.
-	if strings.HasSuffix(query, dotKubernetesZone) && strings.Count(query, ".") < 3 {
```
but I still don't know why we don't care unless they have at least two additional elements.

Reply:
To my knowledge there are no names of interest that would be just `xxx.cluster.local`. You would have at least `xxx.svc.cluster.local`, `xxx.pod.cluster.local` or similar.

Reply:
... and we are using `cluster.local` as the domain which means that queries like `xxx.cluster.local` do arrive quite often.